internal/clients/connect/client_connect_test.go

```diff
@@ -590,3 +591,131 @@ func (s *ConnectClientSuite) TestContentDetails() {
+		events.ServerErrorCode,
+		http_client.NewHTTPError("", "", http.StatusForbidden),
+		nil,
+	)
+	httpClient.On("Get", "/__api__/v1/content/e8922765-4880-43cd-abc0-d59fe59b8b4b", content, lgr).Return(returnErr)
+
+	client := &ConnectClient{
+		client: httpClient,
+	}
+	expectedErr := types.NewAgentError(
+		events.DeploymentFailedCode,
+		errors.New("Cannot deploy content: ID e8922765-4880-43cd-abc0-d59fe59b8b4b - You may need to request collaborator permissions or verify the credentials in use"),
+		nil)
+	err := client.ValidateDeploymentTarget("e8922765-4880-43cd-abc0-d59fe59b8b4b", lgr)
+	aerr, ok := types.IsAgentError(err)
+	s.Equal(ok, true)
+	s.Equal(
+		expectedErr.Message,
+		aerr.Message,
+	)
+}
+
+func (s *ConnectClientSuite) TestValidateDeploymentTargetNotFoundFailure() {
+	lgr := logging.New()
+	content := &ConnectContent{}
+	httpClient := &http_client.MockHTTPClient{}
+
+	// Not Found
+	returnErr := types.NewAgentError(
+		events.ServerErrorCode,
+		http_client.NewHTTPError("", "", http.StatusNotFound),
+		nil,
+	)
+
+	httpClient.On("Get", "/__api__/v1/content/e8922765-4880-43cd-abc0-d59fe59b8b4b", content, lgr).Return(returnErr)
+
+	client := &ConnectClient{
+		client: httpClient,
+	}
+	expectedErr := types.NewAgentError(
+		events.DeploymentFailedCode,
+		errors.New("Cannot deploy content: ID e8922765-4880-43cd-abc0-d59fe59b8b4b - Content cannot be found"),
+		nil)
+	err := client.ValidateDeploymentTarget("e8922765-4880-43cd-abc0-d59fe59b8b4b", lgr)
+	aerr, ok := types.IsAgentError(err)
+	s.Equal(ok, true)
+	s.Equal(
+		expectedErr.Message,
+		aerr.Message,
+	)
+}
+
+func (s *ConnectClientSuite) TestValidateDeploymentTargetUnknownFailure() {
+	lgr := logging.New()
+	content := &ConnectContent{}
+	httpClient := &http_client.MockHTTPClient{}
+
+	// Not Found
+	returnErr := types.NewAgentError(
+		events.ServerErrorCode,
+		http_client.NewHTTPError("", "", http.StatusBadGateway),
+		nil,
+	)
+
+	httpClient.On("Get", "/__api__/v1/content/e8922765-4880-43cd-abc0-d59fe59b8b4b", content, lgr).Return(returnErr)
+
+	client := &ConnectClient{
+		client: httpClient,
+	}
+	expectedErr := types.NewAgentError(
+		events.DeploymentFailedCode,
+		errors.New("Cannot deploy content: ID e8922765-4880-43cd-abc0-d59fe59b8b4b - Unknown error: unexpected response from the server (502)"),
+		nil)
+	err := client.ValidateDeploymentTarget("e8922765-4880-43cd-abc0-d59fe59b8b4b", lgr)
+	aerr, ok := types.IsAgentError(err)
+	s.Equal(ok, true)
+	s.Equal(
+		expectedErr.Message,
+		aerr.Message,
+	)
+}
+
+func (s *ConnectClientSuite) TestValidateDeploymentTargetLockedFailure() {
+	lgr := logging.New()
+	content := &ConnectContent{}
+	queryResult := &ConnectContent{
+		Name:               "",
+		Title:              "",
+		GUID:               "",
+		Description:        "",
+		AccessType:         "",
+		ServiceAccountName: "",
+		DefaultImageName:   "",
+		Locked:             true,
+	}
+	httpClient := &http_client.MockHTTPClient{}
+
```

Comment:
below, I was able to use the mock library's ability to run a function to implement the actual update into the ConnectContent struct. This allowed us to properly unit test the locked content test.